clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp

```diff
@@ -843,7 +864,7 @@ void ModuleDepCollectorPP::addModuleDep(
         !MDC.isPrebuiltModule(Import)) {
```

Comment:
I don't think this handles non-sysroot prebuilt modules. I think that if you have a PCH that depends on a non-sysroot module then this doesn't get tracked.

Reply:
In hindsight, I wish I made this a stacked diff against the real repo.

I have an implementation that addresses this in https://github.com/cyndyishida/llvm-project/commit/272b49206c51c4cc8879c2b539f199c5548fb47a I'd like to merge the current PR and then open a new PR against main for it. But I am open to other suggestions to make the review easier. 

That implementation adds some overhead for two reasons.
1. It deserializes out more content from prebuilt pcm files. 
2. When a module path is fully realized, it traverses down into the leaf dependency and updates whether they are in the sysroot. 

I measured this overhead on a large-ish objc project that uses pch's and toy examples against the macos sdk. I did not observe a noticeable difference.

Reply:
I think that's fine as long as the current patch says that pre built modules means it's not in sysroot, as I think that's a small addition and keeps it conservatively correct.

Reply:
Added a test for this, modified from the working patch. https://github.com/llvm/llvm-project/pull/130634/files#diff-06c3b647f63ed03628260e6d6c1e679990852e445d03dd60d4466248c1f32a5a